drivers: stepper: drv8424 use step_dir common code

faxe1008 opened the pull request:
drivers: stepper: drv8424 use step_dir common code

Now that the common code also has the timing source API available adjust the driver to leverage it.
CC @Irockasingranite

faxe1008 committed:
drivers: stepper: step_dir: Add fallback for dual-edge-step

Adds a fallback for the dual-edge-step property in the step dir common
code. Without this drivers using the common code would have to declare the
dual-edge-step property so it can default to false when not set, even if
the IC does not support adjusting it.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 committed:
drivers: stepper: drv8424: Use step_dir common code

Adapt the drv8424 driver to use the common step dir interface.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

jilaypandya commented on drivers/stepper/step_dir/step_dir_stepper_common.h:
```diff
@@ -45,7 +45,7 @@ struct step_dir_stepper_common_config {
 		.step_pin = GPIO_DT_SPEC_GET(node_id, step_gpios),                                 \
 		.dir_pin = GPIO_DT_SPEC_GET(node_id, dir_gpios),                                   \
-		.dual_edge = DT_PROP(node_id, dual_edge_step),                                     \
+		.dual_edge = DT_PROP_OR(node_id, dual_edge_step, false),                           \
```
This commit is not required, i suppose :)

faxe1008 replied:
It's needed otherwise the dual-edge-step property needs to be part of either the driver dts or the stepper-controller dts binding. You might not want to do this if the IC does not support this setting. But if the property is not present in dts compilation with just `DT_PROP` will fail.

This way only drivers that have that setting can declare it in their own dts, otherwise its just false.

jilaypandya reviewed (commented):
LGTM

faxe1008 committed:
drivers: stepper: step_dir: Fix actual position increment

Address issue where the timing source is not stopped upon step completion
and not adjusting the actual position after performing a step.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 commented:
Found some issue in the common step_dir code that I missed, revealed by the drv8424 test suite.
Very good thing to have @Irockasingranite :^)
